From: Eric Dumazet <eric.dumazet@gmail.com>
Date: Wed, 23 Nov 2011 07:09:32 +0000
Subject: [08/23] net: treewide use of RCU_INIT_POINTER

commit 2cfa5a0471fef43fda0b7bd87e3a5e4dbadb7809 upstream.

rcu_assign_pointer(ptr, NULL) can be safely replaced by
RCU_INIT_POINTER(ptr, NULL)

(old rcu_assign_pointer() macro was testing the NULL value and could
omit the smp_wmb(), but this had to be removed because of compiler
warnings)

Signed-off-by: Eric Dumazet <eric.dumazet@gmail.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
[bwh: Restrict to drivers/net/team/]
---
--- a/drivers/net/team/team_mode_activebackup.c
+++ b/drivers/net/team/team_mode_activebackup.c
@@ -56,7 +56,7 @@ drop:
 static void ab_port_leave(struct team *team, struct team_port *port)
 {
 	if (ab_priv(team)->active_port == port)
-		rcu_assign_pointer(ab_priv(team)->active_port, NULL);
+		RCU_INIT_POINTER(ab_priv(team)->active_port, NULL);
 }
 
 static int ab_active_port_get(struct team *team, void *arg)
